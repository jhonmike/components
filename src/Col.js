import React from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'

import styles from './col.css'

const prepareProps = (props) => {
  const {
    className,
    ...otherProps
  } = props

  const classMap = {
    xs: 'xs',
    sm: 'sm',
    md: 'md',
    lg: 'lg',
    xl: 'xl',
    xsOffset: 'xs-offset',
    smOffset: 'sm-offset',
    mdOffset: 'md-offset',
    lgOffset: 'lg-offset',
    xlOffset: 'xl-offset'
  }

  const sizes = Object.keys(props)
    .filter(key => classMap[key])
    .map(key => {
      const classe = typeof props[key] === 'number'
        ? styles[`${classMap[key]}-${props[key]}`]
        : styles[classMap[key]]

      delete otherProps[classMap[key]]
      return classe
    })

  return {
    ...otherProps,
    className: classnames(...sizes, className)
  }
}

const Col = props => {
  const {
    className,
    children,
    ...otherProps
  } = prepareProps(props)

  return (
    <div {...otherProps} className={className}>
      {children}
    </div>
  )
}

Col.propTypes = {
  className: PropTypes.string,
  xs: PropTypes.oneOfType([PropTypes.number, PropTypes.bool]),
  sm: PropTypes.oneOfType([PropTypes.number, PropTypes.bool]),
  md: PropTypes.oneOfType([PropTypes.number, PropTypes.bool]),
  lg: PropTypes.oneOfType([PropTypes.number, PropTypes.bool]),
  xl: PropTypes.oneOfType([PropTypes.number, PropTypes.bool]),
  xsOffset: PropTypes.number,
  smOffset: PropTypes.number,
  mdOffset: PropTypes.number,
  lgOffset: PropTypes.number,
  xlOffset: PropTypes.number,
  children: PropTypes.node
}

export default Col
