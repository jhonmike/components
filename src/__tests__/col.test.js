import React from 'react'
import renderer from 'react-test-renderer'
import Col from '../Col'

describe('Col Component', () => {
  it('renders responsive component', () => {
    const tree = renderer
      .create(
        <Col xs={12} sm={4} md={3} lg={2}>
          Col
        </Col>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders col with offset', () => {
    const tree = renderer.create(<Col xsOffset={6} xs={6} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders col with auto with', () => {
    const tree = renderer.create(<Col md />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
