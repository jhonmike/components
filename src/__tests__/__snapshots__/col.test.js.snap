// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col Component renders col with auto with 1`] = `
<div
  className="md"
/>
`;

exports[`Col Component renders col with offset 1`] = `
<div
  className="xs-offset-6 xs-6"
  xsOffset={6}
/>
`;

exports[`Col Component renders responsive component 1`] = `
<div
  className="xs-12 sm-4 md-3 lg-2"
>
  Col
</div>
`;
